server/blob/blob-cassandra/src/test/java/org/apache/james/blob/cassandra/cache/CachedBlobStoreTest.java

```diff
@@ -0,0 +1,154 @@
+    public void shouldNotCacheWhenNotDefaultBucketName() {
+        BlobId blobId = Mono.from(testee().save(TEST_BUCKERNAME, EIGHT_KILOBYTES, SIZE_BASED)).block();
+
+        SoftAssertions.assertSoftly(ignored -> {
+            assertThat(Mono.from(cache.read(blobId)).blockOptional()).isEmpty();
+            assertThat(Mono.from(backend.readBytes(TEST_BUCKERNAME, blobId)).block()).containsExactly(EIGHT_KILOBYTES);
+        });
+    }
+
+    @Test
+    public void shouldNotCacheWhenDefaultBucketNameAndBigByteDataAndSizeBase() {
+        BlobId blobId = Mono.from(testee().save(DEFAULT_BUCKERNAME, TWELVE_MEGABYTES, SIZE_BASED)).block();
+
+        SoftAssertions.assertSoftly(ignored -> {
+            assertThat(Mono.from(cache.read(blobId)).blockOptional()).isEmpty();
+            assertThat(Mono.from(backend.readBytes(DEFAULT_BUCKERNAME, blobId)).block()).containsExactly(TWELVE_MEGABYTES);
+        });
+    }
+
+    @Test
+    public void shouldSavedBothInCacheAndBackendWhenSizeBase() {
+        BlobId blobId = Mono.from(testee().save(DEFAULT_BUCKERNAME, EIGHT_KILOBYTES, SIZE_BASED)).block();
+
+        SoftAssertions.assertSoftly(soflty -> {
+            assertThat(Mono.from(cache.read(blobId)).block()).containsExactly(EIGHT_KILOBYTES);
+            assertThat(Mono.from(backend.readBytes(DEFAULT_BUCKERNAME, blobId)).block()).containsExactly(EIGHT_KILOBYTES);
+        });
+    }
+
+    @Test
+    public void shouldSavedBothInCacheAndBackendWhenHighPerformance() {
+        BlobId blobId = Mono.from(testee().save(DEFAULT_BUCKERNAME, EIGHT_KILOBYTES, HIGH_PERFORMANCE)).block();
+
+        SoftAssertions.assertSoftly(soflty -> {
+            assertThat(Mono.from(cache.read(blobId)).block()).containsExactly(EIGHT_KILOBYTES);
+            assertThat(Mono.from(backend.readBytes(DEFAULT_BUCKERNAME, blobId)).block()).containsExactly(EIGHT_KILOBYTES);
+        });
+    }
+
+    @Test
+    public void shouldNotCacheWhenLowCost() {
+        BlobId blobId = Mono.from(testee().save(DEFAULT_BUCKERNAME, EIGHT_KILOBYTES, LOW_COST)).block();
+
+        SoftAssertions.assertSoftly(soflty -> {
+            assertThat(Mono.from(cache.read(blobId)).blockOptional()).isEmpty();
+            assertThat(Mono.from(backend.readBytes(DEFAULT_BUCKERNAME, blobId)).block()).containsExactly(EIGHT_KILOBYTES);
+        });
+    }
+
+    @Test
+    public void shouldRemoveBothInCacheAndBackendWhenDefaultBucketName() {
+        BlobId blobId = Mono.from(testee().save(DEFAULT_BUCKERNAME, EIGHT_KILOBYTES, SIZE_BASED)).block();
+
+        SoftAssertions.assertSoftly(ignored -> {
+            assertThatCode(Mono.from(testee().delete(DEFAULT_BUCKERNAME, blobId))::block)
+                .doesNotThrowAnyException();
+            assertThat(Mono.from(cache.read(blobId)).blockOptional()).isEmpty();
+            assertThatThrownBy(() -> Mono.from(backend.readBytes(DEFAULT_BUCKERNAME, blobId)).block())
+                .isInstanceOf(ObjectNotFoundException.class);
+        });
+    }
```

Comment:
Add a test demonstrating you don't cache empty streams, add a comment explaining that it's technical limitation you had with Pushback input stream.